### ISSUE TITLE
[Hackathon 2024][Niobium][Android] Usage of contentDescription should be required

### ISSUE THREAD

**Issue by tdeffains** [green-code-initiative/ecoCode-challenge#111]
**Rule title**

Accessibility: Provide `contentDescription` attribute for each UI element

**Language and platform**

XML - Android

**Rule description**

For accessibility purpose each UI element should be described for visual disabled user who will navigate through the application using Google TalkBack. See [Google documentation](https://developer.android.com/guide/topics/ui/accessibility/apps#describe-ui-element).
This description is provided by adding `contentDescription` attribute to the UI element.

Bad code:
`
<ImageView
        android:id="@+id/image_id"
        android:src="@drawable/ic_image"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        />
`

Good code:
`
<ImageView
        android:id="@+id/image_id"
        android:src="@drawable/ic_image"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        **android:content_description="@string/image_description"**
        />
`

**Rule short description**

`contentDescription` should be provided for each UI element.

**Rule justification**

Accessibility has become a major concern

**Severity / Remediation Cost**

Estimate the severity and remediation cost of your issue.

Severity must be one of blocker, critical, major, minor (see https://docs.sonarsource.com/sonarqube/latest/extension-guide/adding-coding-rules/#default-severities). Add the justification for your choice.

Remediation cost must be one of trivial, easy, medium, major, high, complex (see https://docs.sonarsource.com/sonarqube/latest/extension-guide/adding-coding-rules/#evaluation-of-the-remediation-cost to choose the must adapted to your rule). Add the justification for your choice.

**Implementation principle**

Indicate here what we will have to detect during the static analysis of the source code with the issue.

**Comment by olegoaer** [comment]
Accessibility is out of the scope of ecoCode (ecology only)
